pallets/capacity/src/lib.rs

```diff
@@ -367,6 +378,26 @@ pub mod pallet {
 			});
 			Ok(())
 		}
+
+		/// Sets the epoch period length (in blocks).
+		///
+		/// # Requires
+		/// * Root Origin
+		///
+		/// ### Errors
+		/// - Returns `Error::MaxEpochLengthExceeded` if `length` is greater than T::MaxEpochLength.
+		#[pallet::call_index(3)]
+		#[pallet::weight(T::WeightInfo::unstake())]
+		pub fn set_epoch_length(
+			origin: OriginFor<T>,
+			length: T::BlockNumber,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+			ensure!(length <= T::MaxEpochLength::get(), Error::<T>::MaxEpochLengthExceeded);
+
+			EpochLength::<T>::set(length);
+			Ok(())
```

Comment:
Should an event be dispatched here?

Reply:
You beat me to it @enddynayn

Reply:
For a different story. @shannonwells, do we have a way to calculate the start of the next epoch? Also, when does this change take effect?